### DISCUSSION TITLE
Default responder

### DISCUSSION THREAD

**Discussion by nhorton** [ElMassimo/oj_serializers#14] [Q&A]
Hi all,
I have been trying to figure out how to turn off jbuilder as the default responder for JSON and use OJ. Anyone know the incantation?

**Comment by ElMassimo** [comment]
Hi Noah!

Would you share an example of the controller action, and how you would expect it to work?

Normally, if you use `render json: AlbumSerializer.one(album)`, `jbuilder` shouldn't interfere.

**Comment by nhorton** [comment]
@ElMassimo - sorry for the slow reply and thanks for the great gem!

Its not about a conflict. It is that with jbuilder, you can have an action like:
```
def show
  @mything = blah
end
```

If it was an HTML request, `show.html.erb` will get used. And if it was JSON, then `show.json.jbuilder` will get used. But with oj_serializers out of the box, you have to have an explicit `respond_to` block to handle the rendering with the serializer. That is what I want to fix - get the oj_serializer used as the default mime handling responder.

**Reply by ElMassimo** [reply]
This gem does not attempt to infer a serializer for a specific object, as from a maintenance perspective, using serializers implicitly makes it really hard to detect dead code in the long run.

---

If you would like to add this behavior to your app, you could define a `default_render` method in `ApplicationController`:

```ruby
def default_render
  if @json
	render json: "#{@json.class.name}Serializer".constantize.render(@json)
  else
    super
  end
end
```

you would then use this by setting `@json` in your controller actions:

```ruby
def show
  @json = blah
end
```

You could even play with having one serializer per controller action, with:

```ruby
render json: "#{controller_name.classify}::#{action_name.classify}Serializer".constantize.render(@json)
```

Notice that serializers don't lend themselves to the "implicit render" pattern, as you would need to somehow specify which object should be passed to the serializer—unlike views which have access to all instance variables.

**Comment by nhorton** [comment]
Fair enough. I appreciate the sample code!
